verilog/CST/identifier_test.cc

```diff
@@ -178,5 +178,31 @@ TEST(GetIdentifierTest, IdentifierUnpackedDimensions) {
   }
 }
 
+// TODO
+TEST(FindAllSymbolIdentifierTest, VariousIds) {
+  // TODO
+  const std::pair<absl::string_view,
+                  std::set<absl::string_view>> kTestCases[] = {
+                    // TODO more cases
+      {"function foo(); endfunction", {"foo"}},
+      {"function myclass::foo(); endfunction", {"myclass", "foo"}},
+      {"task goo(); endtask", {"goo"}},
+      {"task fff::goo(); endtask", {"fff", "goo"}},
+  };
+  for (const auto test : kTestCases) {
+    VerilogAnalyzer analyzer(test.first, "");
+    ASSERT_OK(analyzer.Analyze());
+    const auto& root = analyzer.Data().SyntaxTree();
+    const auto symb_ids = FindAllSymbolIdentifierLeafs(*root);
```

Comment:
Recommend using the `TestVerilogSyntaxRangeMatches` form in the examples above, which will not only check string contents, but also exact byte-offset ranges of text.  Your test case data becomes arrays of `SyntaxTreeSearchTestCase`, which lets you tag the substrings of interest in-line.

A `std::set<string_view>` won't be able to discern duplicate identifiers in the actual set, but the `SyntaxTreeSearchTestCase` form will be able to.